Code Editor linting docs (#2750)
hashicorp/design-system: website/docs/components/code-editor/partials/guidelines/guidelines.md

If you wish to create custom examples using the Code Editor, we publish all of the relevant syntax highlighting variables in the [HDS Components v2.0](https://www.figma.com/design/iweq3r2Pi8xiJfD9e6lOhF/HDS-Components-v2.0?node-id=67166-37020&t=gWdKy44MzTP4cTRo-1) library. However, due to the number of languages supported by the component, the color variables use a generic naming schema (e.g., cyan, red, purple) to remain as agnostic as possible when being applied to different languages.
For more details around syntax, visit the [specifications](/components/code-editor?tab=specifications).

## Linting

> The Code Editor supports linting for JSON using [CodeMirror6](https://codemirror.net/examples/lint/). This feature highlights all errors with an underline and an icon next to the line number. Each icon has a tooltip explaining the error on the associated line. To view all alerts in the editor, users can open a dialog using Ctrl-Shift-m (Cmd-Shift-m on macOS).

Reviewer: ```suggestion
The Code Editor supports linting for JSON using [CodeMirror6](https://codemirror.net/examples/lint/). This feature highlights all errors with an underline and an icon next to the line number. Each icon has a tooltip explaining the error on the associated line. To view all alerts in the editor, open the alert dialog using Ctrl-Shift-m (Cmd-Shift-m on macOS).
```
[Nit] Use more active language

> The Code Editor supports linting for JSON using [CodeMirror6](https://codemirror.net/examples/lint/). This feature highlights all errors with an underline and an icon next to the line number. Each icon has a tooltip explaining the error on the associated line. To view all alerts in the editor, users can open a dialog using Ctrl-Shift-m (Cmd-Shift-m on macOS).

Reviewer 2: [suggestion] remove the duplicative content that's also available below with a visual example.

```suggestion
The Code Editor supports linting for JSON using [CodeMirror6](https://codemirror.net/examples/lint/). This feature highlights all errors with an underline and an icon next to the line number. Each icon has a tooltip explaining the error on the associated line. 
```

Reviewer 3: Thanks for catching that! I've updated the images and text now

!!! Info 

> This component is only available in Ember and is not supported in Figma.

Reviewer: ```suggestion
Linting is only available in the Ember component and is not supported in Figma.
```
[Suggestion] Make banner more specific to linting so it's not taken out of context.

!!!

> ![](/assets/components/code-editor/codeeditor-linting-preview-tooltip.png)

Reviewer: ```suggestion
![Code editor with linting errors. On hover of an icon indicating an error on a line, a tooltip displays details on the linting error.](/assets/components/code-editor/codeeditor-linting-preview-tooltip.png)
```
[Issue] Add alt text

> When linting is enabled, the CodeEditor will have a minimum height set by default to avoid the alert dialog covering all the content within the editor when opened. The minimum height for the CodeEditor with linting is 160px. The linting alert dialog is always 80px in height.

Reviewer: ```suggestion
When linting is enabled, the Code Editor will have a minimum height set by default to avoid the alert dialog covering all the content within the editor when opened. The minimum height for the Code Editor with linting is 160px. The linting alert dialog is always 80px in height.
```
[Nit] Align to how component is named elsewhere.

> ![](/assets/components/code-editor/codeeditor-linting-preview.png)

Reviewer: ```suggestion
![Code editor with linting errors. A red x icon is added to each line with errors, and the text of the line has a red underline.](/assets/components/code-editor/codeeditor-linting-preview.png)
```
[Issue] Add alt text

> To view all alerts in the editor, users can open a dialog using Ctrl-Shift-m (Cmd-Shift-m on macOS).

Reviewer: ```suggestion
To view all alerts in the editor, open the alert dialog using Ctrl-Shift-m (Cmd-Shift-m on macOS).
```
[Nit] Use more active language

![Code Editor with linting errors. There is a dialog on top of the bottom half of the Code Editor with several alerts that can be dismissed. The dialog does not block the user from interacting with the Code Editor content.](/assets/components/code-editor/codeeditor-linting-preview-dialog.png)

> If you require linting for addition languages [contact the HDS team](/about/support).

Reviewer: ```suggestion
If you require linting for additional languages, [contact the Design Systems Team](/about/support).
```
[Nit] Align to how we phrase this elsewhere
